src/File.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace React\Http;
+
+use React\Stream\ReadableStreamInterface;
+
+class File implements FileInterface
+{
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var string
+     */
+    protected $filename;
+
+    /**
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * @var ReadableStreamInterface
+     */
+    protected $stream;
+
+    /**
+     * @param string $name
+     * @param string $filename
+     * @param string $type
+     * @param ReadableStreamInterface $stream
+     */
+    public function __construct($name, $filename, $type, ReadableStreamInterface $stream)
+    {
+        $this->name = $name;
+        $this->filename = $filename;
+        $this->type = $type;
+        $this->stream = $stream;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
```

Comment:
Not always. When you upload small files that fit within a 4KB chunk you generally have a good change of parsing that file in one go. If you go beyond that the parser has no clue when that file will end and won't know until it reached the end. Now the deferred stream catches those and stores them in a buffer but the parser it self just emits a file with a readable stream for it's contents. So at the moment of finding the file it will emit the file and start processing it, without a clue how big it is at that time.